measuring success (#7472)
microsoft/vscode-docs: api/extension-guides/chat.md

});
```

## Measuring success

> We recommend measuring the success of your participant by telemetry logging `Unhelpful` user feedback events, and by telemetry logging the total number of requests your participant handled. Thus, an initial participant success metric can be defined as: `unhelpful_feedback_count / total_requests`.

Reviewer: ```suggestion
We recommend that you measure the success of your participant by adding telemetry logging for `Unhelpful` user feedback events, and for the total number of requests that your participant handled. An initial participant success metric can then be defined as: `unhelpful_feedback_count / total_requests`.
```

```typescript
const logger = vscode.env.createTelemetryLogger({
     // telemetry logging implementation goes here
});

cat.onDidReceiveFeedback((feedback: vscode.ChatResultFeedback) => {
    if (logger.isUsageEnabled) {
        // Log chat result feedback to be able to compute the success matric of the participant
        logger.logUsage('chatResultFeedback', {
            kind: feedback.kind
        });
    }
});
```

> Any other user interaction with your chat response should be measured as a positive metric (e.g. user clicking on a button that your response generated). Measuring success with telemetry is crucial when working with AI since it is an indeterministic technology. Measure and iteratively improve your participant to ensure a good user experience.

Reviewer: ```suggestion
Any other user interaction with your chat response should be measured as a positive metric (for example, the user selecting a button that was generated in a chat response). Measuring success with telemetry is crucial when working with AI, since it is an nondeterministic technology. Measure and iteratively improve your participant to ensure a good user experience.
```
